src/nimblepkg/version.nim

```diff
@@ -2,9 +2,12 @@
 # BSD License. Look at license.txt for more info.
 
 ## Module for handling versions and version ranges such as ``>= 1.0 & <= 1.5``
-import strutils, tables, hashes, parseutils
+import json
+import common, strutils, tables, hashes, parseutils
+
 type
-  Version* = distinct string
+  Version* {.requiresInit.} = object
+    version: string
```

Comment:
This way the `Version` type can only be created by the `newVersion` procedure which does validation. Distinct type does not provide such guarantee because someone can write `Version("foo")` for example.